js/controller.js

```diff
@@ -262,8 +270,40 @@ OO.plugin("Html5Skin", function (OO, _, $, W) {
       }
     },
 
+    setIPadVideoPosition: function() {
+      if (Utils.isIPad()) {
+        var videoHeight = this.state.mainVideoElement[0].videoHeight;
+        var videoWidth = this.state.mainVideoElement[0].videoWidth;
+
+        if (videoHeight/videoWidth > this.state.playerHeight/this.state.playerWidth) {
+          this.state.mainVideoElement.css(InlineStyle.iPadVerticalVideoStyle);
+        }
+        else {
+          this.state.mainVideoElement.css(InlineStyle.iPadHorizontalVideoStyle);
+        }
+        this.state.iPadVideoPositionSet = true;
+      }
+    },
+
+    onVideoControllerVideoElementLostFocus: function() {
+      if (Utils.isIPad() && this.state.iPadVideoPositionSet == true) {
+        this.state.iPadVideoPositionSet = false;
+      }
+    },
+
+    onVideoControllerVideoElementInFocus: function() {
+      if (Utils.isIPad() && this.state.iPadVideoPositionSet == false) {
+        //hiding the video before it is positioned correctly by setIPadVideoPosition
+        this.state.mainVideoElement.css({"visibility":"hidden"});
+      }
+    },
+
     onPlaying: function() {
       // pause/resume of Ad playback is handled by different events => WILL_PAUSE_ADS/WILL_RESUME_ADS
+      if (Utils.isIPad() && this.state.iPadVideoPositionSet == false) {
```

Comment:
Couple comments:
1. this.state.iPadVideoPositionSet == false inside IF condition can be refactored to !this.state.iPadVideoPositionSet
2. I would recommend to move isIpad and iPadVideoPositionSet check inside the setIPadVideoPosition()